slack-conversation.c

```diff
@@ -22,8 +61,22 @@ static void conversations_list_cb(SlackAccount *sa, gpointer data, json_value *j
 		return;
 	}
 
-	for (unsigned i = 0; i < chans->u.array.length; i++)
-		conversation_update(sa, chans->u.array.values[i]);
+	gboolean get_history = purple_account_get_bool(sa->account, "get_history", FALSE);
+
+	for (unsigned i = 0; i < chans->u.array.length; i++) {
+		json_value *chan_json = chans->u.array.values[i];
+		conversation_update(sa, chan_json);
+
+		gboolean is_im = json_get_prop_boolean(chan_json, "is_im", FALSE);
+		gboolean is_mpim = json_get_prop_boolean(chan_json, "is_mpim", FALSE);
+		/* Automatically fetch history for all direct message
+		   channels. We assume that other channels will fetch history
+		   separately as part of joining them. */
+		if (get_history && (is_im || is_mpim)) {
+			const char *chan_id = json_get_prop_strptr(chan_json, "id");
+			slack_api_call(sa, get_latest_history_cb, NULL, "conversations.info", "channel", chan_id, NULL);
```

Comment:
Is this call necessary?  Do the `last_read` and `unread_count` fields not come back in the `conversations.list` response?  I can check.

Reply:
I had the exact same thought and I searched quite a lot for this actually. But it doesn't look like the `conversations.list` call returns anything useful in this context, so we have to query each one. Makes me wonder what the official client does.